torch/_dynamo/variables/functions.py

```diff
@@ -341,6 +341,17 @@ def call_function(
             ]:
                 with torch._dynamo.side_effects.allow_side_effects_under_checkpoint(tx):
                     return super().call_function(tx, args, kwargs)
+        elif self.fn is torch._functorch.autograd_function.vmapify_autograd_function:
```

Comment:
just wondering, why didn't you do something like:

UserDefinedFunction(torch._functorch.autograd_function.vmapify_autograd_function).call_function(tx, args) 

The current solution in this PR looks fine to me though

Reply:
Yeah, the default inlining approach is better, but I ran into a few unsupported Dynamo features while handling the following case:
https://github.com/pytorch/pytorch/blob/e68f5087d808654ba5741e158dbea703a7f879ef/torch/_functorch/autograd_function.py#L502-L511

This includes issues like constructing `NestedUserFunctionVariable` without a source, among others. Since I’d like to keep this PR focused on tracing the vmapped autograd function rather than addressing broader issues, I decided to go with this approach for now.

That said, I’m happy to revisit this later and migrate it to the inlining approach as a follow-up. I’ll add a TODO comment here to track it.

Reply:
TODO sounds fine